opsgenie/resource_opsgenie_role.go

```diff
@@ -0,0 +1,220 @@
+		Read:   handleNonExistentResource(resourceOpsGenieCustomUserRoleRead),
+		Update: resourceOpsGenieCustomUserRoleUpdate,
+		Delete: resourceOpsGenieCustomUserRoleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"rolename": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"extended_role": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"user", "observer", "stakeholder",
+				}, false),
+			},
+			"granted_rights": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice(validRights, false),
+				},
+				Set: schema.HashString,
+			},
+			"disallowed_rights": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice(validRights, false),
+				},
+				Set: schema.HashString,
+			},
+		},
+	}
+}
+
+func flattenSet(input *schema.Set) []string {
+	output := make([]string, 0)
+
+	if input == nil {
+		return output
+	}
+
+	for _, v := range input.List() {
+		output = append(output, v.(string))
+	}
+	return output
+}
+
+func resourceOpsGenieCustomUserRoleCreate(d *schema.ResourceData, meta interface{}) error {
+
+	client, err := custom_user_role.NewClient(meta.(*OpsgenieClient).client.Config)
+
+	if err != nil {
+		return err
+	}
+
+	UserRoleName := d.Get("rolename").(string)
+	ExtendedUserRole := d.Get("extended_role").(string)
+	GrantedRights := flattenSet(d.Get("granted_rights").(*schema.Set))
+	DisallowedRights := flattenSet(d.Get("disallowed_rights").(*schema.Set))
+
+	log.Printf("[INFO] Creating OpsGenie custom user role '%s'", UserRoleName)
+	result, err := client.Create(context.Background(), &custom_user_role.CreateRequest{
+		Name:             UserRoleName,
+		ExtendedRole:     custom_user_role.ExtendedRole(ExtendedUserRole),
+		GrantedRights:    GrantedRights,
+		DisallowedRights: DisallowedRights,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(result.Id)
+
+	return resourceOpsGenieCustomUserRoleRead(d, meta)
+}
+
+func resourceOpsGenieCustomUserRoleRead(d *schema.ResourceData, meta interface{}) error {
+	client, err := custom_user_role.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	UserRoleName := d.Get("rolename").(string)
+
+	log.Printf("[INFO] Reading OpsGenie custom role '%s'", UserRoleName)
+
+	usrRole, err := client.Get(context.Background(), &custom_user_role.GetRequest{
+		Identifier:     UserRoleName,
+		IdentifierType: custom_user_role.Name,
+	})
+	if err != nil {
+		return err
+	}
+
+	d.Set("username", usrRole.Name)
+	d.Set("full_name", usrRole.ExtendedRole)
+	d.Set("role", usrRole.GrantedRights)
+	d.Set("locale", usrRole.DisallowedRights)
```

Comment:
I dont understand here ?

Reply:
My bad, typo
I wonder why the tests did not failt, because username, full_name, role, locale is not present in the schema of role resource.